doc/configuration.rst

```diff
@@ -411,6 +411,35 @@ hive resources and 1 mysql resource:
 Note that it was not necessary to specify the 1 for mysql here, but it
 is good practice to do so when you have a fixed set of resources.
 
+.. _retcode-config:
+
+[retcode]
+----------
+
+Configure return codes for the luigi binary. In the case of multiple return
```

Comment:
It adds complexity (for users), but I think it's justified since real-world scheduling is complex, and whoever uses luigi in a big setting should actively think about the corner cases. Keep in mind that:

 * People might already rely on the current behaviour (exit code 1 on unhandled exceptions).
 * You want different return codes in different contexts.
   * Sometimes you simply want the "run until you succeed" behaviour that is the default (exit code 0 for failures/missingdata/alreadyrunning). 
   * Sometimes you want to run a command line, and feel confident that if I get exit code 0, my root task is in fact complete (then you would configure all exit codes).
   * If you use Jenkins as a cron-invoker, you would like task_failed to have a dedicated error code, causing "red" in Jenkins. Meanwhile already running or missing data exit codes should indicate "pending", or indicating your Jenkins dependencies mismatch what's configured in luigi.

That said. I would be thrilled to see this being simplified (convention over configuration), but I don't want to render it useful for only 50% of the users who want exit codes.

> Perhaps declare in documentation a "reserved" range of return codes that Luigi might define later, the rest being available for application-specific usage.

I'm not sure what you mean. You basically want to enforce that one can only choose exit codes within the range [0, 4] or something and that [5..255] are reserved for luigi itself?